test/lib/rules/no-node-env-ssr.js

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+const { RuleTester } = require('eslint');
+
+const { ESLINT_TEST_CONFIG } = require('../shared');
+const rule = require('../../../lib/rules/no-node-env-in-ssr');
+
+const tester = new RuleTester(ESLINT_TEST_CONFIG);
+
+tester.run('no-node-env-in-ssr', rule, {
+    valid: [
+        {
+            code: `
+                import { LightningElement } from 'lwc';
+                import tmplA from './a.html';
+
+                export default class Foo extends LightningElement {
+                  connectedCallback() {
+                    // we can't use process.env.NODE_ENV here
+                  }
+                  renderedCallback() {
+                    if (process.env.NODE_ENV === 'development') {
+                      console.log('test');
+                    }
+                  }
+                  bar() {
+                    if (process.env.NODE_ENV === 'development') {
+                      console.log('test');
+                    }
+                  }
+                }
+            `,
+        },
+        {
+            code: `
+                import { LightningElement } from 'lwc';
+                import tmplA from './a.html';
+
+                export default class Foo extends LightningElement {
+                  connectedCallback() {
+                    // we can't use process.env.NODE_ENV here
+                  }
+                  renderedCallback() {
+                    this.bar();
+                  }
+                  bar() {
+                    if (process.env.NODE_ENV === 'development') {
+                      console.log('test');
+                    }
+                  }
+                }
+            `,
+        },
+        {
+            code: `
+                import { LightningElement } from 'lwc';
+                import tmplA from './a.html';
+
+                export default class Foo extends LightningElement {
+                  connectedCallback() {
+                    // we can't use process.env.NODE_ENV here
+                  }
+                  bar() {
+                    doSomething(process.emv.NODE_ENV);
+                  }
+                }
+            `,
+        },
+    ],
+    invalid: [
```

Comment:
Do a check for use in constructor? In a super's constructor? In class but outside a function?

Reply:
I agree the first & last would be good test cases to add. I don't think the `super`'s constructor is handled by "does it run in SSR"-helpers right now, largely because the superclass is likely in another file and eslint operates on each file discretely. So I'd recommend omitting that test case in this PR.